### ISSUE TITLE
msui on docker ImportError: libGL.so.1: cannot open shared object file

### ISSUE THREAD

**Issue by ReimarBauer** [Open-MSS/MSS#362]
**[Original report](https://bitbucket.org/wxmetvis/mss/issue/362) by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

```
#!text

docker run -i -t dreimark/mss:1.7.3 /bin/bash
(base) root@f171abfc3995:/# mss
Traceback (most recent call last):
  File "/opt/conda/bin/mss", line 7, in <module>
    from mslib.msui.mss_pyui import main
  File "/opt/conda/lib/python3.6/site-packages/mslib/msui/mss_pyui.py", line 49, in <module>
    from mslib.msui.mss_qt import ui_mainwindow as ui
  File "/opt/conda/lib/python3.6/site-packages/mslib/msui/mss_qt.py", line 36, in <module>
    from fslib.fs_filepicker import getSaveFileName, getOpenFileName, getExistingDirectory
  File "/opt/conda/lib/python3.6/site-packages/fslib/fs_filepicker.py", line 35, in <module>
    from PyQt5 import QtWidgets, QtCore
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

minor becaus new platform was initially only for server setup

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

we have to install in the docker setup

apt-get install libgl1-mesa-glx

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Next problem

```
#!text

# mss
WARNING:root:*** NAppy is not available. You will not be able to read NASA Ames files. ***
INFO: MSS Version: 1.7.3
INFO: Python Version: 3.6.5 |Anaconda, Inc.| (default, Apr 29 2018, 16:14:56) 
[GCC 7.2.0]
INFO: Platform: Linux-4.4.0-133-generic-x86_64-with-debian-9.5 (('64bit', ''))
INFO: Launching user interface...
This application failed to start because it could not find or load the Qt platform plugin "xcb"
in "".

Available platform plugins are: eglfs, minimal, minimalegl, offscreen, vnc, xcb.

Reinstalling the application may fix this problem.
Aborted (core dumped)
```

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

* apt-get install python-xpyb
* apt-get install libx11-xcb1
* apt-get install libxi6


```
#!text

mss
WARNING:root:*** NAppy is not available. You will not be able to read NASA Ames files. ***
INFO: MSS Version: 1.7.3
INFO: Python Version: 3.6.5 |Anaconda, Inc.| (default, Apr 29 2018, 16:14:56) 
[GCC 7.2.0]
INFO: Platform: Linux-4.4.0-133-generic-x86_64-with-debian-9.5 (('64bit', ''))
INFO: Launching user interface...
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
```

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

```
#!text

xhost +local:docker
docker run -ti --rm -e DISPLAY=$DISPLAY -v /tmp/.X11-unix/:/tmp/.X11-unix dreimark/mss:1.7.3  /bin/bash
mss
```

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

![Bildschirmfoto von »2018-08-21 09-19-11«.png](https://bitbucket.org/repo/MBp6Xa/images/2705981564-Bildschirmfoto%20von%20%C2%BB2018-08-21%2009-19-11%C2%AB.png)

not soo bad, but where are the fonts?

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

after

apt-get install xfonts*

all fonts seen, but which is needed

at least this font 

 * apt-get install xfonts-scalable

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

text input is blocked, mouse works.

this is because this is not exposed, solution on

https://stackoverflow.com/questions/50789172/libinput-in-a-docker-container

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

docker run -ti --rm --device /dev/input/event3 -e DISPLAY=$DISPLAY -v /tmp/.X11-unix/:/tmp/.X11-unix dreimark/mss:1.7.3  /bin/bash

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

docker image setup adjusted to run also the mss ui, fixes #362

**Comment by ReimarBauer** [comment]
**Original comment by Reimar Bauer (Bitbucket: [ReimarBauer](https://bitbucket.org/ReimarBauer), GitHub: [ReimarBauer](https://github.com/ReimarBauer)).**

----------------------------------------

Merged in ReimarBauer/mss/stable (pull request #572)

docker image setup adjusted to run also the mss ui, fixes #362
